src/rp2_common/hardware_gpio/include/hardware/gpio.h

```diff
@@ -401,7 +401,8 @@ void gpio_set_dormant_irq_enabled(uint gpio, uint32_t events, bool enabled);
  *
  * \param gpio GPIO number
  * \param events Bitmask of events to clear. See \ref gpio_set_irq_enabled for details.
-  *
+ *
+ * \note \ref gpio_set_irq_enabled_with_callback calls this function before the callback.
```

Comment:
`gpio_set_irq_enabled_with_callback` doesn't directly call `gpio_acknowledge_irq`. Did you mean something like "The default GPIO IRQ-handler automatically calls this function before calling any callback-function" ?

Reply:
Sorry yes, I meant to imply that callbacks set with `gpio_set_irq_enabled_with_callback` don't need to acknowledge the IRQ as it's done before the user callback runs.  I've reworded it, hopefully it reads better now.